clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -2364,14 +2364,6 @@ Expected<SmallVector<StringRef>> linkAndWrapDeviceFiles(
         HasNonSYCLOffloadKinds = true;
     }
 
-    // Write any remaining device inputs to an output file.
```

Comment:
This seems to be a community code: https://github.com/llvm/llvm-project/blame/main/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp#L939
 
 I would suggest to drop our own copy within `HasSYCLOffloadKind` instead to reduce divergence with the upstream.

Reply:
"`HasNonSYCLOffloadKinds` branch" builds similar vector differently. I think this is the reason to sink `InputFiles` initialization into if-conditions.

Reply:
> It modifies `InputFiles`, but the code to initialize seems to be absolutely the same. In that case it is not clear to me as to why we can't re-use the community code here.

They are different. Please compare:
https://github.com/intel/llvm/blob/sycl/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp#L2376-L2383 and https://github.com/intel/llvm/blob/sycl/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp#L2464-L2475.

> Shadowing variables is not a good idea

That's the reason I remove it. If we go with your proposal, we still have to use shadowing due to the difference I mentioned above.

Reply:
> That's the reason I remove it. If we go with your proposal, we still have to use shadowing due to the difference I mentioned above.

Isn't better to fix shadowing in our own code which was added on top of the upstream instead of dropping the upstream code?

Reply:
If you have any suggestions, I'm all ears.

Reply:
> If you have any suggestions, I'm all ears.

#16926 is an alternative proposal. I would love to see even more unification, but I can't quickly propose anything as it requires more investigation

Reply:
Your proposal doesn't work. I left a comment in https://github.com/intel/llvm/pull/16926.

Reply:
Talked to @AlexeySachkov offline: we agreed to merge this PR as-is and Alexey will make more clean-up changes in #16926.